plugins/out_es/es_conf.c

```diff
@@ -153,8 +157,32 @@ struct flb_elasticsearch *flb_es_conf_create(struct flb_output_instance *ins,
             flb_es_conf_destroy(ctx);
             return NULL;
         }
+        flb_plg_debug(ctx->ins, "extracted cloud_host: '%s'", cloud_host);
+
+        flb_plg_debug(ctx->ins,
+                      "check whether extracted cloud_host '%s' contains a port",
+                      cloud_host);
+        cloud_host_char = strtok_r(cloud_host, ":", &saveptr);
+        flb_plg_debug(ctx->ins, "cloud_host without port: '%s'", cloud_host_char);
+
+        cloud_port_char = strtok_r(NULL, ":", &saveptr);
+        flb_plg_debug(ctx->ins, "extracted cloud_port_char: '%s'", cloud_port_char);
+
+        if (cloud_port_char != NULL) {
+            cloud_host_port = (int)strtol(cloud_port_char, (char **)NULL, 10);
+            flb_plg_debug(ctx->ins, "converted cloud_port_char to port int: '%i'", cloud_host_port);
+        }
+
+        if (cloud_host_port == NULL) {
```

Comment:
`cloud_host_port` is `int`. I think it should not be NULL.
Should it be `cloud_port_char` ?

Reply:
Changed. Now it checks whether `cloud_host_port` is `0`.